src/Adapter/MSTest.TestAdapter/MSTestSettings.cs

```diff
@@ -262,32 +264,49 @@ public static void PopulateSettings(MSTestSettings settings)
     /// <param name="logger"> The logger for messages. </param>
     /// The discovery context that contains the runsettings.
     /// </param>
-    internal static void PopulateSettings(IDiscoveryContext? context, IMessageLogger? logger)
+    internal static void PopulateSettings(IDiscoveryContext? context, IMessageLogger? logger, IConfiguration? configuration)
     {
-        RunConfigurationSettings = RunConfigurationSettings.PopulateSettings(context);
-
-        if (context?.RunSettings == null || StringEx.IsNullOrEmpty(context.RunSettings.SettingsXml))
+        if (configuration != null && context?.RunSettings != null)
         {
-            // This will contain default adapter settings
-            CurrentSettings = new MSTestSettings();
-            return;
+            throw new InvalidOperationException(Resource.DuplicateConfigurationError);
         }
 
-        MSTestSettings? aliasSettings = GetSettings(context.RunSettings.SettingsXml, SettingsNameAlias, logger);
-
-        // If a user specifies MSTestV2 in the runsettings, then prefer that over the v1 settings.
-        if (aliasSettings != null)
+        if (context?.RunSettings != null && !StringEx.IsNullOrEmpty(context.RunSettings.SettingsXml))
```

Comment:
Is it the `!StringEx.IsNullOrEmpty(context.RunSettings.SettingsXml)` needed? 
If `context?.RunSettings != null` isn't enough?

Reply:
it was like that in old code, so i think it was for a reason

Reply:
Simplification looks fine to me:

```suggestion
        if (!StringEx.IsNullOrEmpty(context?.RunSettings.SettingsXml))
```